nwb_conversion_tools/basesortingextractorinterface.py

```diff
@@ -37,24 +37,8 @@ def run_conversion(self, nwbfile: NWBFile, metadata: dict, stub_test: bool = Fal
                        write_ecephys_metadata: bool = False):
         if 'UnitProperties' not in metadata:
             metadata['UnitProperties'] = []
-        if write_ecephys_metadata and 'Ecephys' in metadata:
-            n_channels = max([len(x['data']) for x in metadata['Ecephys']['Electrodes']])
-            recording = se.NumpyRecordingExtractor(timeseries=np.array(range(n_channels)), sampling_frequency=1)
-            se.NwbRecordingExtractor.add_devices(
-                recording=recording,
-                nwbfile=nwbfile,
-                metadata=metadata
-            )
-            se.NwbRecordingExtractor.add_electrode_groups(
-                recording=recording,
-                nwbfile=nwbfile,
-                metadata=metadata
-            )
-            se.NwbRecordingExtractor.add_electrodes(
-                recording=recording,
-                nwbfile=nwbfile,
-                metadata=metadata
-            )
+        if write_ecephys_metadata:
+            add_ecephys_metadata(nwbfile, metadata)
```

Comment:
The core part of -40 to -57 now occurs in multiple parts of the code, making it easier to simply have an auxiliary util function (+40 to +41) for adding it to the nwbfile when/if needed